api/v1beta1/controlplane_types.go

```diff
@@ -38,6 +38,12 @@ type PlacementSpec struct {
 	Replicas int `json:"replicas,omitempty"`
 }
 
+// InterconnectSpec defines the desired state of AMQInterconnect
+type AMQInterconnectSpec struct {
```

Comment:
I would probably just called this `InterconnectSpec` since the object name is `Interconnect` not `AMQInterconnect` (and is a downstream name)

Reply:
> I would probably just called this `InterconnectSpec` since the object name is `Interconnect` not `AMQInterconnect` (and is a downstream name)

Ack, I will make this change.